File: C++/10866.cpp
#include<iostream>
#include<queue>
#include<deque>
using namespace std;

int main() {
	deque<int> dq;
	string cmd;
	int param, n;
	cin >> n;
	while (n--) {
		cin >> cmd;
		if (cmd == "push_front") {
			cin >> param;
			dq.push_front(param);
		}
		else if (cmd == "push_back") {
			cin >> param;
			dq.push_back(param);
		}
		else if (cmd == "pop_front") {
			if (dq.empty()) {
				cout << -1 << endl;
			}
			else {
				cout << dq.front() << endl;
				dq.pop_front();
			}
		}
		else if (cmd == "pop_back") {
			if (dq.empty()) {
				cout << -1 << endl;
			}
			else {
				cout << dq.back() << endl;
				dq.pop_back();
			}
		}
		else if (cmd == "size") {
			cout << dq.size() << endl;
		}
		else if (cmd == "empty") {
			cout << dq.empty() << endl;
		}
		else if (cmd == "front") {
			if (dq.empty()) {
				cout << -1 << endl;
			}
			else {
				cout << dq.front() << endl;
			}
		}
		else if (cmd == "back") {
			if (dq.empty()) {
				cout << -1 << endl;
			}
			else {
				cout << dq.back() << endl;
			}
		}
	}
}

File: Python/10866.py
import sys

N = int(sys.stdin.readline())
d = []

for x in range(N):
    comnd = sys.stdin.readline().rstrip()
    # print('comnd:', comnd)

    if comnd[:10] == 'push_front':
        d.insert(0, comnd.split()[-1])
    elif comnd[:9] == 'push_back':
        d.append(comnd.split()[-1])
    elif comnd == 'pop_front':
        if d:
            print(d.pop(0))
        else:
            print(-1)
    elif comnd == 'pop_back':
        if d:
            print(d.pop(-1))
        else:
            print(-1)
    elif comnd == 'size':
        print(len(d))
    elif comnd == 'empty':
        if d:
            print(0)
        else:
            print(1)
    elif comnd == 'front':
        if d:
            print(d[0])
        else:
            print(-1)
    elif comnd == 'back':
        if d:
            print(d[-1])
        else:
            print(-1)
    # print('d:', d)

